### DISCUSSION TITLE
Location selection doesn't work

### DISCUSSION THREAD

**Discussion by HodaSadatJafari** [openwisp/django-loci#119] [Q&A] [answered]
Hi,
I cloned the project and run it locally, everything seems to work fine except one part. When I want to add a new device, it asks me to select a location, either create a new location or select from existing location. When I select the "Existing" option, it shows me "location" item which I can select from predefined locations. When I select one location from the list, on the main page it opens all the related fields from the location and fills all the values, like the picture bellow:

![image](https://github.com/user-attachments/assets/f0b4de6f-8312-43f2-8cc2-7e9fc3f25e88)

But sometimes this doesn't work. It means when I select a location, nothing happens and it doesn't show me the location fields and their values, like the picture bellow:

![image](https://github.com/user-attachments/assets/19248624-a819-428a-8a4c-e750b61b9f09)

I don't understand why sometimes the "select location" works and sometimes doesn't work. 
For example right now, I have run the program locally and when I open the admin page on the Firefox the "select location" works and  when open it with Chrome the "select location" doesn't work, but sometimes it is vice versa!
Also, I have deployed the project on a AWS EC2, in this case the "select location" doesn't work in any situations.

I would appreciate your help,
Hoda

**Comment by nemesifier** [comment]
It may be because some Javascript line fails. Please check the Javascript browser console (developer tools) to see if there's any JS error log.

**Reply by HodaSadatJafari** [reply]
I already checked the JS, I get 200 for all the JS
![image](https://github.com/user-attachments/assets/7146b5d9-0807-4bff-b276-efdbe4ed0be7)
I though maybe there is something related to channels...
I only see this error in the console:
![image](https://github.com/user-attachments/assets/29b8a127-1809-41e5-8ad6-c40306bc780d)

**Reply by nemesifier** [reply]
Please check the console tab.

**Reply by HodaSadatJafari** [reply]
In the console tab, I only see this:
![image](https://github.com/user-attachments/assets/e5f31b2a-748b-46ff-bbf9-2727cdbdaa80)

**Reply by nemesifier** [reply]
It seems the image was not uploaded correctly.

Can you upload it again?

Can you also provide a list of steps which allow to replicate the issue you're having?

**Reply by HodaSadatJafari** [reply]
Updated the image.
I just cloned the program, installed the dependencies and run the program. But sometimes when I run the program the select location works and sometimes doesn't work!! I still don't know why and what it depends to that it doesn't work.

**Reply by nemesifier** [reply]
@HodaSadatJafari the image you uploaded is showing the network tab, I am asking you to show the console tab when the unexpected behaviour happens.

There must be something causing this, if you can find out what it is and how to replicate the behavior, let us know.

**Reply by HodaSadatJafari** [reply]
I thought console on the bottom you mean.
Here you go:
![image](https://github.com/user-attachments/assets/63ef423d-adc6-4827-b323-607328ae10de)

**Reply by nemesifier** [reply]
Filter only errors, and make sure to capture the screenshot when the unexpected behavior happens.

**Reply by HodaSadatJafari** [reply]
I don't see any error when I click on the select location. 
But when I refresh the add device page I get this error, I don't know if it is related or not:
![image](https://github.com/user-attachments/assets/5dace28a-5da6-4d58-9048-659e13d9db07)

**Reply by HodaSadatJafari** [reply]
BTW I have integrated the django-loci in our project and deployed it on a server. You can access it via: http://52.28.90.134/admin/
with user: admin 
and password: admin@@admin
Go to "add vehiclestatus" and then select a location from existing list: http://52.28.90.134/admin/fleets/vehiclestatus/add/
You will see nothing happens and it doesn't load the location

**Reply by nemesifier** [reply]
Check if you have this change deployed: https://github.com/openwisp/django-loci/commit/25651595ca8d67fe81853a79bf0000b0ba5cc1fd.

**Reply by HodaSadatJafari** [reply, marked as answer]
No I didn't have.
I had installed django-loci before with command: `pip install django-loci`
Now I install it with this command: `pip install https://github.com/openwisp/django-loci/tarball/master`
It fixed the JS and CSS files. Why the installations are different?
This fixed the problem. 
Thank you very much :)

**Reply by nemesifier** [reply]
We are using the latest development version in our OpenWISP deployments and we haven't release a new version on pypi yet.
We will be releasing in the coming weeks.
